Automattic/pocket-casts-ios
# Download Issues: Cancel download tasks when no episode matches

On app-launch, this prunes download tasks which have no matching episode.

## To test

* Set a breakpoint at both of the cancels to see that they are run
* Kill the app by stopping in Xcode (force quitting will cause the tasks to be cancelled)
* Run the following shell script:
```bash
#!/bin/bash

SQL_COMMAND="DELETE FROM SJEpisode WHERE downloadTaskId IN (SELECT downloadTaskId FROM SJEpisode WHERE downloadTaskId IS NOT NULL LIMIT 10);"
sqlite3 "$(xcrun simctl get_app_container booted au.com.shiftyjelly.podcasts data)/Library/Application Support/Pocket Casts/podcast_newDB.sqlite3" "$SQL_COMMAND"

echo "Deleted first 10 episodes with downloadTaskId NOT NULL"
```
* Make sure breakpoints are hit
* Ensure episodes in podcast are no longer downloading
* Try downloading again and make sure they work

## Checklist
